unit-test/background/pixel.es6.js

```diff
@@ -0,0 +1,60 @@
+
+const pixel = require('../../shared/js/background/pixel.es6')
+const url = 'https://improving.duckduckgo.com/t/'
+const getURLTestCases = [
+    {
+        'pixelName': 'ep',
+        'result': url + 'ep'
+    },
+    {
+        'pixelName': '',
+        'result': 'undefined'
+    },
+    {
+        'pixelName': 'undefined',
+        'result': 'undefined'
+    }
+]
+const concatParamsTestCases = [
+    {
+        'params': [],
+        'partialResult': ''
+    },
+    {
+        'params': [
+            'param1'
+        ],
+        'partialResult': '_param1'
+    },
+    {
+        'params': [
+            'param1',
+            'param2'
+        ],
+        'partialResult': '_param1_param2'
+    }
+]
+
+describe('pixel.getURL()', () => {
+    getURLTestCases.forEach((test) => {
+        beforeEach(function () {
+            Object.keys(test).forEach((key) => {
+                test[key] = (test[key] === 'undefined') ? undefined : test[key]
+            })
+        })
+
+        it(`should return ${test.result} as a result given the pixelName: ${test.pixelName}`, () => {
+            let result = pixel.getURL(test.pixelName)
+            expect(result).toBe(test.result)
+        })
+    })
+})
+
+describe('pixel.concatParams()', () => {
+    concatParamsTestCases.forEach((test) => {
+        it(`should return a string containing ${test.partialResult} and for params: ${test.params}`, () => {
+            let result = pixel.concatParams(test.params)
+            expect(result).toMatch(test.partialResult)
+        })
+    })
+})
```

Comment:
Can we add a specific test that covers the entirety of `fire`? You'd need to set up spies for `load.url()` and `settings.getSetting()`. I made a handy reusable spy for `settings` - ping me when you're about if you want me to show you it!

Reply:
Hey @andrey-p, I added other test cases for the existing unit tests. I don't think it's worth adding tests for fire()? All the data used to send the URL is returned from other functions that we're unit testing already. I mean, besides actually loading the URL, what do you think should be tested here?
I wouldn't hold the PR for it, but let me know if you disagree :)